src/components/ebay-snackbar-dialog/component.js

```diff
@@ -0,0 +1,22 @@
+const DEFAULT_TIMEOUT_LENGTH = 6000;
+module.exports = {
+    onInput(input) {
+        if (input.open === true || input.open === false) {
+            this.state.open = input.open;
+            clearTimeout(this.timeout);
+        }
+        if (input.open !== false) {
+            clearTimeout(this.timeout);
+            const timeout = setTimeout(() => {
+                this.emit('close');
```

Comment:
One enhancement to make is that the dialog should never close while keyboard focus or mouse is inside.

Reply:
Okay so I made progress on this, but I need a new selector. Right now I am using the CSS one and that doesn't seem very stable. See the component.js in the snackbar dir [commit here](https://github.com/eBay/ebayui-core/pull/1338/commits/ed58d0bf87fbb3168b7ba0c51c189eb53f42187c)